### PR TITLE
librados,osd,mon: remove traces of CEPH_OSDMAP_FULL

### PR DESCRIPTION
**Pull request by tchaikov**
## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard backend`
- `jenkins test docs`
- `jenkins render docs`

</details>

### CODE REVIEW

**Comment by xiexingguo** [comment]
removing the unnecessary code is always my preferred option, so my :+1:  for this.
I doubt there might be so old cluster-wide flag tests need to be fixed at first, so let's wait and see what qa would say about this

**Comment by tchaikov** [comment]
failure tracked at https://tracker.ceph.com/issues/42077 

jenkins test make check

File: src/client/Client.cc [hunk 1]
```diff
@@ -2460,7 +2460,7 @@ void Client::_handle_full_flag(int64_t pool)
   {
     Inode *inode = i->second;
     if (inode->oset.dirty_or_tx
-        && (pool == -1 || inode->layout.pool_id == pool)) {
+        && inode->layout.pool_id == pool) {
```
Comment (liewegas) [inline src/client/Client.cc]: I think this means that if we use the octopus client to talk to a pre-luminous (?) cluster, which doesn't set per-pool full flags, we'll misbehave?

If that's what we want to do, we need to document it.  I'm a bit unsure we should drop this code, though.

Comment (tchaikov) [reply src/client/Client.cc]: @liewegas i dropped this commit from this PR.

**Comment by liewegas** [comment]
Before we merge this, we should document the compatibility the other way around, too: which version of the client (userspace and kernel) first supported the per-pool FULL flag?

**Comment by tchaikov** [comment]
@liewegas documented in manpage and PendingReleaseNotes. could you lemme know if anything missing?

File: PendingReleaseNotes [hunk 2]
```diff
@@ -151,6 +151,25 @@
...
+  of the cluster back in the Hammer release, if the OSD map is marked ``full``
+  all write operations will be blocked until this flag is removed. In the
+  Infernalis release, we introduced the per-pool full/nearfull flags to track
+  the status for a finer-grained control, but only the clients supported this
+  flag, so they will hold the write operations if either the cluster-wide
+  ``full`` flag or the per-pool ``full`` flag is set. This was a compromise,
+  as we needed to support both the cluster with and without per-pool ``full``
+  flags support. But this practically defeated the purpose of introducing
+  the per-pool flags. So, in the Mimic release, the new flag finally took the
+  place of cluster-wide ``full`` flag in the Mimic release, as the monitor
```
Comment (xiexingguo) [inline PendingReleaseNotes]: dup `in the Mimic release`

Comment (tchaikov) [reply PendingReleaseNotes]: revised and repushed.

**Comment by liewegas** [comment]
@idryomov what kernel version do the per-pool flags correspond to?

I wonder if we should adjust the OSDMap.cc get_required_features to explicitly map to I or J (or an equivalent feature) so that the requirement is explicit and enforced.  (This would also means we could start cleaning out the compat behavior for everything else below that line.)

**Comment by idryomov** [comment]
@liewegas Per-pool flags field was introduced in kernel 3.9.  Support for per-pool CEPH_POOL_FLAG_FULL came in kernel 4.7.

**Comment by tchaikov** [comment]
added kernel version to the newly added section in PendingReleaseNotes

**Pull request merged**